src/modules/windowwalker/app/Window Walker/Components/OpenWindows.cs

```diff
@@ -94,23 +94,32 @@ public bool WindowEnumerationCallBack(IntPtr hwnd, IntPtr lParam)
         {
             Window newWindow = new Window(hwnd);
 
-            if (windows.Select(x => x.Title).Contains(newWindow.Title))
+            uint toolWindow = (uint)InteropAndHelpers.ExtendedWindowStyles.WS_EX_TOOLWINDOW;
+            uint appWindow = (uint)InteropAndHelpers.ExtendedWindowStyles.WS_EX_APPWINDOW;
+            InteropAndHelpers.GetWindowCmd owner = InteropAndHelpers.GetWindowCmd.GW_OWNER;
+            int windowLong = InteropAndHelpers.GetWindowLong(hwnd, InteropAndHelpers.GWL_EXSTYLE);
+
+            if ((newWindow.IsWindow && newWindow.Visible && InteropAndHelpers.GetWindow(hwnd, owner) != null &&
+                (((windowLong & toolWindow) != toolWindow) || ((windowLong & appWindow) == appWindow)) &&
+                InteropAndHelpers.GetProp(hwnd, "ITaskList_Deleted") == IntPtr.Zero &&
+                newWindow.ClassName != "Windows.UI.Core.CoreWindow" /*&& !newWindow.IsWindowCloaked() //hides all cloaked apps*/)

```

Comment:
Same comment as the one above. Move these into boolean properties in `Window` to both simplify the `if` statement and also make it reusable elsewhere.

Reply:
Done!